### PR TITLE
Update IDFilter.pxd

### PR DESCRIPTION
**Pull request by timosachsenberg**
# Description

Add a missing method to IDFilter

# Checklist:
- [X] Make sure that you are listed in the AUTHORS file
- [X] Add relevant changes and new features to the CHANGELOG file
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] New and existing unit tests pass locally with my changes
- [X] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

### CODE REVIEW

File: src/pyOpenMS/pxds/IDFilter.pxd [hunk 1]
```diff
@@ -91,7 +91,11 @@ cdef extern from "<OpenMS/FILTERING/ID/IDFilter.h>" namespace "OpenMS":
         void filterHitsByScore(MSExperiment& experiment, double peptide_threshold_score, double protein_threshold_score) nogil except +
 
         void keepNBestHits(MSExperiment& experiment, Size n) nogil except +
+        
+        void keepBestPerPeptide(libcpp_vector[PeptideIdentification]& peptides, bool ignore_mods, bool ignore_charges, Size nr_best_spectrum) #wrap-doc:Filters PeptideHits from PeptideIdentification by keeping only the best peptide hits for every peptide sequence
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/IDFilter.pxd:95]:
```suggestion
        void keepBestPerPeptide(libcpp_vector[PeptideIdentification]& peptides, bool ignore_mods, bool ignore_charges, Size nr_best_spectrum) nogil except + #wrap-doc:Filters PeptideHits from PeptideIdentification by keeping only the best peptide hits for every peptide sequence
```

File: src/pyOpenMS/pxds/IDFilter.pxd [hunk 2]
```diff
@@ -91,7 +91,11 @@ cdef extern from "<OpenMS/FILTERING/ID/IDFilter.h>" namespace "OpenMS":
...
 
+        void keepBestPerPeptidePerRun(libcpp_vector[ProteinIdentification]& prot_ids, libcpp_vector[PeptideIdentification]& peptides, bool ignore_mods, bool ignore_charges, Size nr_best_spectrum) #wrap-doc:Filters PeptideHits from PeptideIdentification by keeping only the best peptide hits for every peptide sequence on a per run basis
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/IDFilter.pxd:97]:
```suggestion
        void keepBestPerPeptidePerRun(libcpp_vector[ProteinIdentification]& prot_ids, libcpp_vector[PeptideIdentification]& peptides, bool ignore_mods, bool ignore_charges, Size nr_best_spectrum) nogil except + #wrap-doc:Filters PeptideHits from PeptideIdentification by keeping only the best peptide hits for every peptide sequence on a per run basis
```

**Review by oliveralka — APPROVED** [review]
Looks good - I think "nogil except" missing in both cases.

**Comment by timosachsenberg** [comment]
I wonder why these don't need to be wrapped as static methods usually are...

**Comment by oliveralka** [comment]
That is actually a pretty good question. We should test the nightly if the IDFilter works without any issues or if they have to be wrapped as static via `#wrap-attach:IDFilter`.

**Pull request merged**